### ISSUE TITLE
Color codes not stripped from records with colorize=False

### ISSUE THREAD

**Issue by gazpachoking** [Delgan/loguru#198]
When adding a function sink with `colorize=False` the `message` attribute of the record does not have the color codes stripped. Here's a short example showing the issue I mean:

```python
from loguru import logger


def test_sink(message):
    assert "<red>" not in message  # This is fine
    assert "<red>" not in message.record["message"]  # This fails


logger.add(test_sink, colorize=False)
logger.opt(ansi=True).info("<red>something</>")
```

This is a problem for example when sending log records over the network, as in this recipe. https://loguru.readthedocs.io/en/stable/resources/recipes.html#sending-and-receiving-log-messages-across-network-or-processes

On the other side of the network, the record message still contains color codes, but we have no way of transmitting that the message should be processed with the `.opt(ansi=True` anymore. Perhaps rather than (or maybe in addition to) `.opt()` modifying the logger, the options should be attached to the log record?

**Comment by gazpachoking** [comment]
This might also complicate the proposed solution in #197 where color codes are ignored in format arguments.

**Comment by gazpachoking** [comment]
One possible solution is to just change this line:
https://github.com/Delgan/loguru/blob/d4851c4f8798ebb392cea3b13caa2d78f5644fc2/loguru/_handler.py#L140

to
```python
str_record.record = formatter_record
```

Not sure if there are other implications for that.

**Comment by Delgan** [comment]
Thanks once again for the report!

It's part of the "design" that the same `record` instance is passed to each sink, so I don't think I want to replace `str_record.record = record` with `formatter_record` as each sink would receive a different `dict` in this case. Attaching `opt=True` to the `record` seems unappealing from my point of view, as such option is more related to the way of formatting the `record`, while the `record` should only contain information about the context of the logging call. In addition, there are many other options in `opt()`, and I prefer to avoid getting special treatment to one of them and not the others.

I've been thinking a little bit about that. My conclusion is that color tags should purely and simply be removed from the `message` attribute. My point is that the color markups are not actually part of the message string, they are more like "metadata" which is used independently by each sink (to colorize or not the message). The process to convert or strip color tags is internal to Loguru, so it makes little sense to output something containing color tags, they should be for inputs only.

I quite like the idea that the `message` only contain the actual `message` stripped from the color tags. What do you think?

Now, the question is how to propagate the colors through the socket... I guess on the server side you could either bypass the formatting with `logger.opt(raw=True).log(level, formatted_message)`. Alternatively, on the client side, use `format=lambda _: "{message}"` which implies the `formatted_message` on the server side will be equals to the colored `record["message"]`. Here are the few workarounds that come to mind. :confused:

**Comment by gazpachoking** [comment]
Hmm. In my case, I have sinks on both the client and server side, that may or may not support colors. My understanding is that your proposed workarounds wouldn't handle that case, or am I wrong?

**Comment by Delgan** [comment]
You're right, it will not work well with multiple sinks on both server and client side. :confused: 

In any case, it doesn't seem possible to correct this considering #197: color tags and logging `args` should not be formatted at the same time, but it's complicated for the client to propagate them separately to the server.

I don't think there is any proper solution I could implement. In my opinion, we have to accept the fact that color tags in the message cannot be transmitted through the network.

Here is another possible workaround, though: you could force `colorize=True` on the client side, and manually strip ansi codes in your sink (instead of relying on `colorize`) if needed in your server sinks (using [`ansistrip`](https://pypi.org/project/ansistrip/) for example).

```python
# server.py
from ansistrip import ansi_strip

logger.add(sink_supporting_colors)
logger.add(lambda m: sink_not_supporting_colors(ansi_strip(m)))
```

**Comment by gazpachoking** [comment]
> I've been thinking a little bit about that. My conclusion is that color tags should purely and simply be removed from the `message` attribute.

Yeah, stripping the codes from message seems fine. Colored log messages over network would have been nice, but not worth bending over backwards for. We can try some of your proposed workarounds in the future if colored messages become a larger priority in our project at any point.

**Comment by Delgan** [comment]
Fixed in `v0.4.1`: the `record["message"]` will now always contain the stripped message.
